src/ui/screens/onboarding/ProfileScreen.tsx

```diff
@@ -40,8 +47,18 @@ type Props = DispatchProps & StateProps;
 
 export const ProfileScreen = (props: Props) => {
     const modelHelper = new ReactNativeModelHelper(props.gatewayAddress);
-    const authorImage = getImageSource(props.author.image, modelHelper, defaultImages.defaultUser);
+    const authorImage = getImageSource(props.author.image, modelHelper);
     const isFormFilled = props.author.name !== '' && props.author.name !== defaultAuthor.name;
+    const [ identity, setIdentity ] = React.useState<PrivateIdentity | undefined>(undefined);
+    React.useEffect(() => {
+        Swarm.generateSecureIdentity(generateSecureRandom).then(value => {
+            setIdentity(value);
+            getFallbackUserImage(value.publicKey).then(image => {
+                props.onUpdatePicture(image);
+            });
+        });
+    }, []);
```

Comment:
trying out the hooks API. One downside is that useEffect does not accept an `async` function.